pkg/providers/gcp/provider_redis.go

```diff
@@ -51,114 +54,195 @@ func (rp RedisProvider) GetReconcileTime(r *v1alpha1.Redis) time.Duration {
 
 var _ providers.RedisProvider = (*RedisProvider)(nil)
 
-func (p *RedisProvider) CreateRedis(ctx context.Context, r *v1alpha1.Redis) (*providers.RedisCluster, croType.StatusMessage, error) {
-	logger := p.Logger.WithField("action", "CreateRedis")
+func (rp *RedisProvider) CreateRedis(ctx context.Context, r *v1alpha1.Redis) (*providers.RedisCluster, croType.StatusMessage, error) {
+	logger := rp.Logger.WithField("action", "CreateRedis")
 	logger.Infof("reconciling redis %s", r.Name)
-	// handle provider-specific finalizer
-	if err := resources.CreateFinalizer(ctx, p.Client, r, DefaultFinalizer); err != nil {
-		return nil, "failed to set finalizer", err
+	var statusMessage string
```

Comment:
I'm not overly bothered by this but is there any benefit to explicitly defining the statusMessage early? In the case of no errors it's unused and there is only a single assignment in each case on error the same as before. But if I understand correctly the original approach avoids creating the errMsg string at all on successful reconcile?

Reply:
Yes, that's correct. I've updated the statusMessage variables to not be declared before they're needed 👍🏻